folding/src/standard_config.rs

```diff
@@ -0,0 +1,471 @@
+//! This module offers an standard implementation of [FoldingConfig] supporting
+//! many use cases
+use crate::{
+    expressions::FoldingColumnTrait, instance_witness::Witness, FoldingConfig, FoldingEnv,
+    Instance, Side,
+};
+use kimchi::{circuits::gate::CurrOrNext, curve::KimchiCurve};
+use memoization::ColumnMemoizer;
+use poly_commitment::{commitment::CommitmentCurve, srs};
+use std::{fmt::Debug, hash::Hash, marker::PhantomData, ops::Index};
+
+#[derive(Clone, Default)]
+/// Default type for when you don't need structure
+pub struct EmptyStructure<G: KimchiCurve>(PhantomData<G::ScalarField>);
+
+impl<G: KimchiCurve, Col> Index<Col> for EmptyStructure<G> {
+    type Output = Vec<G::ScalarField>;
+
+    fn index(&self, _index: Col) -> &Self::Output {
+        panic!("shouldn't reach this point, as this type only works with witness-only constraint systems");
+    }
+}
+
+/// An standard folding config that supports:
+/// `G`: any curve
+/// `Col`: any column implementing [FoldingColumnTrait]
+/// `Chall`: any challenge
+/// `Sel`: any dynamic selector
+/// `Str`: structures that can be indexed by `Col`, thus implementing `Index<Col>`
+/// `I`: instances (implementing [Instance]) that can be indexed by `Chall`
+/// `W`: witnesses (implementing [Witness]) that can be indexed by `Col` and `Sel`
+/// ```ignore
```

Comment:
Can we avoid ignoring it? So we build all the time with the doc and check it is up-to-date?

Reply:
It is ignored because it isn't working code, it wouldn't even build as some of the types there are not defined.
I can later rewrite some of the examples we have using this config and make them into a doc, but I would prefer to do it in another PR.

Reply:
Sure, not urgent at all. I noticed it in [this comment[(https://github.com/o1-labs/proof-systems/pull/2270/files#r1626589975).